### PR TITLE
Add support for AsyncAPI standard to verify_docs

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

**Comment by alfrunes** [comment]
@lluiscampos I can't assign reviewers here, but could you please take a look?

**Comment by lluiscampos** [comment]
> @lluiscampos I can't assign reviewers here, but could you please take a look?

Fixed now the access right for this repo.

Reviewing :eyes:

File: verify_docs.py [hunk 1]
```diff
@@ -6,22 +8,62 @@
...
-                k = cfg["paths"][i][j]
+            spec = yaml.load(ymlfile, Loader=yaml.SafeLoader)
 
-                if "summary" not in k:
-                    print("Error: summary missing from Swagger doc in: %s " % (i))
-                    errored = True
+        if "openapi" in spec:
+            return verify_swagger(spec)
+        elif "swagger" in spec:
+            return verify_openapi(spec)
```
Comment (lluiscampos) [inline verify_docs.py]: This looks wrong
```suggestion
        if "swagger" in spec:
            return verify_swagger(spec)
        elif "openapi" in spec:
            return verify_openapi(spec)
```

File: verify_docs.py [hunk 2]
```diff
@@ -6,22 +8,62 @@
...
+
+def verify_openapi(spec):
+    errored = False
+    for i in spec["paths"]:
+        for j in spec["paths"][i]:
+            k = spec["paths"][i][j]
+            if "summary" not in k:
+                print("Error: summary missing from OpenAPI doc in: %s " % (i))
+                errored = True
+    return errored
```
Comment (lluiscampos) [inline verify_docs.py]: These are the exact same logic, right? Except for the error message...

Not strong opinion on leaving it as-is if in the future we might have different checks on them. tt just felt so weird (I did check it line by line...).

File: verify_docs.py [hunk 3]
```diff
@@ -6,22 +8,62 @@
...
+
+
+def verify_asyncapi_1(spec):
+    msgs = spec["components"]["messages"]
+    errored = False
+    for name, msg in msgs.items():
+        if "summary" not in msg:
+            print("Error: #components/messages/%s is missing a summary" % name)
```
Comment (lluiscampos) [inline verify_docs.py:59]: To be consistent, something like:
```suggestion
            print("Error: #components/messages/%s from AsyncAPI is missing a summary" % name)
```

File: verify_docs.py [hunk 4]
```diff
@@ -6,22 +8,62 @@
...
+            return verify_swagger(spec)
+        elif "swagger" in spec:
+            return verify_openapi(spec)
+        elif "asyncapi" in spec:
+            api_version = StrictVersion(spec["asyncapi"])
+            if api_version.version[0] == 1:
+                # We currently only support major version 1.
+                return verify_asyncapi_1(spec)
+            else:
+                print("Error: cannot verify openapi version")
```
Comment (lluiscampos) [inline verify_docs.py]: Suggestion:
```suggestion
                print("Error: cannot verify openapi major version %d" % api_version.version[0])
```

File: verify_docs.py [hunk 5]
```diff
@@ -6,22 +8,62 @@
...
+                return True
 
-    return errored
+        else:
+            print("Error: unable to identify API spec kind.")
```
Comment (lluiscampos) [inline verify_docs.py:87]: Suggestion
```suggestion
            print("Error: unable to identify API spec kind for spec %s" % spec)
```

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Good thinking.

Only one thing that needs fixing (lines 51-54), the rest are comments and suggestions.

File: verify_docs.py (hunk 2 above)
Comment (alfrunes) [reply verify_docs.py]: I was thinking of adding logic for checking unused references, but it turned out to be a little more complicated than I initially thought.

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**